### ISSUE TITLE
Integration tests: Version specified in `engines.vscode` (*) is not specific enough

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#147789]
I think these may be new, seeing them in our integration tests (remote) and local:

https://dev.azure.com/monacotools/Monaco/_build/results?buildId=167467&view=logs&j=672276a2-8d3a-5fab-615d-090c51352f92&t=6e7e74bd-a473-53b1-acea-ad5f8c1f1ec9&l=516

https://dev.azure.com/monacotools/Monaco/_build/results?buildId=167467&view=logs&j=672276a2-8d3a-5fab-615d-090c51352f92&t=6a86faf8-ed15-56db-877e-46756fb8e8fa&l=715

```
[9568:0420/172929.807:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.808:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.809:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.827:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.918:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.918:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.918:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.929:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.941:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.941:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.944:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.951:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.956:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.957:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.959:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172929.974:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.036:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.037:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.039:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.047:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.448:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.450:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.452:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.472:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.690:INFO:CONSOLE(29)] "%c INFO color: #33f Starting worker process with pid 6632 (type: fileWatcher, window: 1).", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.711:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/grunt]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.712:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/gulp]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.715:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/jake]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
[9568:0420/172930.729:INFO:CONSOLE(29)] "%c  ERR color: #f33 [/D:/a/_work/1/s/extensions/vscode-colorize-tests]: Version specified in `engines.vscode` (*) is not specific enough. For vscode versions before 1.0.0, please define at a minimum the major and minor desired version. E.g. ^0.10.0, 0.10.x, 0.11.0, etc.", source: vscode-file://vscode-app/d:/a/_work/1/VSCode-win32-x64/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (29)
```

**Comment by alexdima** [comment]
The way we are running these tests causes quite some problems. We are running tests for builtin extensions but we don't identify them as builtin extensions because they are on disk outside of the built vscode (which makes sense, because we don't want to ship tests with vscode). But we now need to do all kinds of workarounds, like allowing proposed API via the command line e.g. https://github.com/microsoft/vscode/issues/146172 , or now we would need to disable or suppress engine validations, which I don't think we really want to do.

@sandy081 Should we have a way to mark the extension currently being developed as builtin? Do you already have a low hanging mechanism for this?

**Comment by sandy081** [comment]
@alexdima How about marking extension under development as builtin if the extension is located under builtin extensions location? I assume these extensions running tests are in the same folder as other builtin extensions?

**Comment by alexdima** [comment]
>  I assume these extensions running tests are in the same folder as other builtin extensions?

No. That is the root problem. We build vscode to e.g. `~/src/VSCode-darwin` and we launch the tests from the vscode checkout location e.g. `~/src/vscode`. So we just see `--extension-development` pointing to a folder outside the builtin extensions. But maybe we could look if it overwrites a builtin extension, it should get the builtin flag set.

**Comment by alexdima** [comment]
Interesting, we're already doing that -- https://github.com/microsoft/vscode/blob/3f557f2d39e4a94a0c6ba133e3058298cab2c2a0/src/vs/workbench/services/extensions/common/extensionsUtil.ts#L39-L40

**Comment by sandy081** [comment]
Got it.

> But maybe we could look if it overwrites a builtin extension, it should get the builtin flag set.

At present we are scanning builtin, user and extensions under development in parallel. This means we have to read builtin extensions first and pass them as input to reading extensions under development. I think this is ok as scanning extensions under development is not a common scenario.

**Comment by alexdima** [comment]
> This means we have to read builtin extensions first and pass them as input to reading extensions under development.

:+1: Nice idea! Would you want to do it? We could then also remove all these ugly arguments of `--enable-proposed-api` from [here](https://github.com/microsoft/vscode/blob/3f557f2d39e4a94a0c6ba133e3058298cab2c2a0/scripts/test-remote-integration.sh#L66).

**Comment by sandy081** [comment]
Sure, will do it.

**Comment by sandy081** [comment]
Fixed. Under dev extensions will take existing extension type if exist. So built in extensions running integration tests in the build will have built-in extension type and therefore we do not need to set `--enable-proposed-api` flag in the integration test scripts. So, I also removed this flag from the scripts.

Here is the build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=169064&view=results with these changes.
